src/System.Private.TypeLoader/src/Internal/Runtime/TypeLoader/TypeLoaderEnvironment.LdTokenResultLookup.cs

```diff
@@ -127,14 +127,17 @@ private struct RuntimeMethodHandleKey : IEquatable<RuntimeMethodHandleKey>
 
             public RuntimeMethodHandleKey(RuntimeTypeHandle declaringType, string methodName, RuntimeSignature signature, RuntimeTypeHandle[] genericArgs)
             {
-                Debug.Assert(genericArgs != null);
```

Comment:
>Assert [](start = 22, length = 6)

Could you keep the assert and change it to check that genericsArgs == null || .length > 0, and a comment explaining that null is used for generic method definitions.